specification/containerservice/Fleet.Management/gate.tsp

```diff
@@ -0,0 +1,204 @@
+  updateRunProperties: UpdateRunGateTargetProperties,
+}
+
+// A wrapper is needed because the union does not create one itself. Without it, properties of
+// all different types would end up in a single `properties` model.
+@doc("Wrapper object around the Update Run Gate Target Properties.")
+@added(Versions.v2025_04_01_preview)
+model UpdateRunGateTargetProperties {
+  @doc("Properties specific to the Update Run.")
+  updateRun: UpdateRunGateTarget;
+}
+
+@doc("The Gate is targeting an Update Run staged rollout.")
+@added(Versions.v2025_04_01_preview)
+model UpdateRunGateTarget {
+  @doc("The name of the Update Run.")
+  @visibility(Lifecycle.Read)
+  @pattern("^[a-z0-9]([-a-z0-9]*[a-z0-9])?$")
+  @minLength(1)
+  @maxLength(50)
+  name: string;
+
+  @doc("The Update Stage of the Update Run.")
+  @visibility(Lifecycle.Read)
+  @pattern("^[a-z0-9]([-a-z0-9]*[a-z0-9])?$")
+  @minLength(1)
+  @maxLength(50)
+  stage: string;
+
+  @doc("The Update Group of the Update Run.")
+  @visibility(Lifecycle.Read)
+  @pattern("^[a-z0-9]([-a-z0-9]*[a-z0-9])?$")
+  @minLength(1)
+  @maxLength(50)
+  group?: string;
+
+  @doc("Whether the Gate is placed before or after the update itself.")
+  @visibility(Lifecycle.Create, Lifecycle.Read)
+  beforeOrAfter: BeforeOrAfter;
+}
+
+@doc("Whether the Gate is placed before or after the target.")
+@added(Versions.v2025_04_01_preview)
+union BeforeOrAfter {
+  string,
+
+  @doc("The Gate is before the target.")
+  Before: "Before",
+
+  @doc("The Gate is after the target.")
+  After: "After",
+}
+
+scalar UpdateRunResourceId
+  extends Azure.Core.armResourceIdentifier<[
+    {
+      type: "Microsoft.ContainerService/fleets/updateRuns",
+    }
+  ]>;
+
+@doc("The resource id of staged rollout that the Gate is targeting.")
+@added(Versions.v2025_04_01_preview)
+union GateTargetId {
+  @doc("The Gate is controlling the staged rollout of an Update Run.")
+  updateRun: UpdateRunResourceId,
+}
+
+@doc("The target that the Gate is controlling, e.g. an Update Run.")
+@added(Versions.v2025_04_01_preview)
+model GateTarget {
+  @doc("The resource id that the Gate is controlling the rollout of.")
+  @visibility(Lifecycle.Create, Lifecycle.Read)
+  id: GateTargetId;
+
+  @doc("The type of the resource being targeted by this Gate.")
+  @visibility(Lifecycle.Create, Lifecycle.Read)
+  type: GateTargetType;
+
+  @doc("Properties specific to the target of this Gate.")
+  @visibility(Lifecycle.Create, Lifecycle.Read)
+  properties: GateTargetProperties;
+}
+
+@doc("The provisioning state of the Gate resource.")
+@lroStatus
+union GateProvisioningState {
+  string,
+  ResourceProvisioningState,
+}
+
+@armResourceOperations
+@added(Versions.v2025_04_01_preview)
+interface Gates {
+  get is ArmResourceRead<Gate>;
+
+  // The create and delete APIs are both internal-only as Gates are created by the Update Run during
+  // its processing. As such, we only expose the update API here.
```

Comment:
let's see how that goes :)
alternative as discussed is to declare it here, but validate the caller.